src/AbstractLinesBeforeNamespaceFixer.php

```diff
@@ -36,13 +36,17 @@ protected function fixLinesBeforeNamespace(Tokens $tokens, $index, $expected)
     {
         // if we've got a <?php, then subtracted the number of new lines it
         // contains from the expected number in the following whitespace
-        if (isset($tokens[$index - 2])) {
-            $opening = $tokens[$index - 2];
-            if ($opening->isGivenKind(T_OPEN_TAG)) {
-                $expected -= substr_count($opening->getContent(), "\n");
+        for ($i = 1; $i <= 2; ++$i) {
+            if (isset($tokens[$index - $i])) {
+                $opening = $tokens[$index - $i];
+                if ($opening->isGivenKind(T_OPEN_TAG)) {
+                    $expected -= substr_count($opening->getContent(), "\n");
```

Comment:
This is assuming that the open tag always ends with `"\n"`, which is not true.

Please add this test case:
```php
array("<?php\n\nnamespace X;", '<?php namespace X;'),
```

Reply:
So, the fixer should also strip the trailing white space in `'<?php '`?

Reply:
>So, the fixer should also strip the trailing white space in '<?php '?

that is not needed IMHO, but the test case is to point something else I think, try;

```php
array("<?php \n\nnamespace X;", '<?php namespace X;'),
```

Reply:
> `array("<?php \n\nnamespace X;", '<?php namespace X;'),`

This passes the test

Reply:
good to hear :) than I ping @julienfalque for his thoughts on it

Reply:
> So, the fixer should also strip the trailing white space in `'<?php '`?

Yes that's what I meant, sorry for being unclear.

I fixed a similar case in `header_comment` in #2630 which got merged so I think we should do it here too.

Reply:
Done